static/src/server.ts

```diff
@@ -428,29 +446,108 @@ function getMapTileResult(
       legendG.appendChild(legendSvg);
       mergedSvg.appendChild(legendG);
 
-      return {
-        svg: getProcessedSvg(mergedSvg),
-        data_csv: mapDataToCsv(chartData.geoJson, chartData.dataValues),
-        srcs: getSources(chartData.sources),
-        legend: "",
-        title: getChartTitle(tileConfig.title, getMapRS(tileProp, chartData)),
-        type: "MAP",
-      };
+      return [
+        {
+          svg: getProcessedSvg(mergedSvg),
+          data_csv: mapDataToCsv(chartData.geoJson, chartData.dataValues),
+          srcs: getSources(chartData.sources),
+          title: getChartTitle(tileConfig.title, getMapRS(tileProp, chartData)),
+          type: "MAP",
+        },
+      ];
     })
     .catch(() => {
       console.log("Failed to get map tile result for: " + id);
       return null;
     });
 }
 
-// Gets the TileResult for a given tile config and supporting information.
+// Get the result for a single ranking unit
+function getRankingUnitResult(
+  tileConfig: TileConfig,
+  rankingGroup: RankingGroup,
+  sv: string,
+  isHighest: boolean
+): TileResult {
+  const rankingHtml = ReactDOMServer.renderToString(
+    getRankingUnit(
+      tileConfig.title,
+      sv,
+      rankingGroup,
+      tileConfig.rankingTileSpec,
+      isHighest
+    )
+  );
+  const style = {
+    "font-family": FONT_FAMILY,
+    "font-size": FONT_SIZE,
+  };
+  const svg = htmlToSvg(rankingHtml, SVG_WIDTH, SVG_HEIGHT, "", style);
+  const processedSvg = getProcessedSvg(svg);
+  return {
+    svg: processedSvg,
+    data_csv: rankingPointsToCsv(rankingGroup.points, rankingGroup.svName),
+    srcs: getSources(rankingGroup.sources),
+    title: getRankingUnitTitle(
+      tileConfig.title,
+      tileConfig.rankingTileSpec,
+      rankingGroup,
+      false,
+      sv
+    ),
+    type: "RANKING",
+  };
+}
+
+// Get the tile results for a ranking tile.
+function getRankingTileResult(
+  id: string,
+  tileConfig: TileConfig,
+  place: NamedTypedPlace,
+  enclosedPlaceType: string,
+  statVarSpec: StatVarSpec[]
+): Promise<TileResult[]> {
+  const tileProp = {
+    id,
+    title: tileConfig.title,
+    place,
+    enclosedPlaceType,
+    statVarSpec,
+    rankingMetadata: tileConfig.rankingTileSpec,
+    apiRoot: CONFIG.apiRoot,
+  };
+  return fetchRankingData(tileProp)
+    .then((rankingData) => {
```

Comment:
consider using `await` here and making this function `async`

Reply:
done